aya-obj/src/obj.rs

```diff
@@ -712,10 +712,30 @@ impl Object {
         Ok(())
     }
 
-    fn parse_program(&self, section: &Section) -> Result<(Program, Function), ParseError> {
-        let prog_sec = ProgramSection::from_str(section.name)?;
-        let name = prog_sec.name().to_owned();
+    fn parse_programs(
+        &mut self,
+        section: &Section,
+        symbols: HashMap<String, Symbol>,
+    ) -> Result<(), ParseError> {
+        let progs = symbols
+            .iter()
+            .map(|(name, symbol)| self.parse_program(section, name, symbol))
+            .collect::<Result<Vec<(Program, Function)>, ParseError>>()?;
```

Comment:
There's no need to collect here. Instead of using map, iterate on
symbols.iter(), and we save an extra allocation + an extra loop

Reply:
Done

Reply:
Done